addons/web_editor/static/src/js/wysiwyg/widgets/link_tools.js

```diff
@@ -75,9 +75,12 @@ const LinkTools = Link.extend({
 
     applyLinkToDom() {
         this._observer.disconnect();
+        const wasObserving = this.options.wysiwyg.odooEditor.isObserverObserving;
         this.options.wysiwyg.odooEditor.observerActive();
         this._super(...arguments);
-        this.options.wysiwyg.odooEditor.observerUnactive();
+        if (!wasObserving) {
+            this.options.wysiwyg.odooEditor.observerUnactive();
```

Comment:
Do the original line (that I merged) actually made sense? https://github.com/odoo/odoo/commit/c29fe9fa8fc8ff4a5e6f042607ef9b3070d1d339

It seems strange now that I re-see them. Your fix of course makes sense but it seems to be a band-aid on a broken system. And if not, it definitely needs comments. I am really surprised I merged the original fix without understand more, it seems already to be a huge band-aid. Should not the observer be active at all times and only disabled for short durations (just the duration of a DOM modification we don't want to track?). Disabling in the start of the link tools and reanabling in their destroy seems very wrong.

Reply:
Indeed, there doesn't seem to be any reason to disable the observer at the start. By removing it, I haven't found any case where a bad behavior occurred and it passes the tests. In the worst case we would add the o_dirty class where before we didn't add it, so it would be saved, which doesn't seem that bad, does it? Could there be a corner case that I could have missed ? If not I think it's indeed better like it is now :+1: